windows-containers/windows-monitoring.md

```diff
@@ -0,0 +1,163 @@
+---
+title: OpenShift Windows Worker Node Monitoring
+authors:
+  - "@ravisantoshgudimetla"
+  - "@aravindhp"
+reviewers:
+  - "@crawford"
+  - "@sdodson"
+  - "@brancz"
+  - "@s-urbaniak"
+  - "@lilic"
+approvers:
+  - "@crawford"
+  - "@sdodson"
+  - "@brancz"
+  - "@s-urbaniak"
+  - "@lilic"
+creation-date: 2019-09-17
+last-updated: 2019-09-18
+status: implementable
+---
+
+# monitoring-of-windows-worker-nodes
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The intent of this enhancement is to enable monitoring of Windows worker nodes
+in OpenShift cluster.
+
+## Motivation
+
+Monitoring is critical to identify issues with nodes, containers running on the
+nodes. The main motivation behind this enhacement is to enable monitoring of
+Windows nodes
+
+### Goals
+
+As part of this enhacement, we plan to do the following:
+* Deploy Windows Management Instrumentation(WMI exporter) onto Windows nodes
+* Upgrade WMI exporter on the Windows Nodes
+* Leverage cluster-monitoring operator that sets up Prometheus, Alertmanager
+  and other components
+
+### Non-Goals
+
+As part of this enhacement, we do not plan to do the following:
+* Create a container image for WMI and let WMI run as a container
+* Integrating WMI with cluster monitoring operator
+* Ship Grafana dashboards for Windows Nodes
+
+## Proposal
+
+The main idea here is to run WMI exporter as a Windows Service and let 
+Prometheus instance which was provisioned as part of OpenShift install to 
+collect data from WMI exporter. Since WMI exporter cannot run as a container
+on the Windows node, there by making it impossible to run it within OpenShift cluster, 
+we need to ensure that Prometheus talks to WMI exporter as an external service.
+
+
+### Implementation Details/Notes/Constraints [optional]
```

Comment:
Is it possible to use the playbook tooling to build a golden image that a customer can boot and just join the cluster?

The playbook as defined seems to co-mingle changes to the host state along with the cluster itself.  It would be nice if there was a clearer separation.  I would imagine many customers will want to create a baked Windows image that they use to join a cluster.

Reply:
>Is it possible to use the playbook tooling to build a golden image that a customer can boot and just join the cluster?

Sure but that would be needed in general and not as part of the monitoring enhancement. I suggest that we keep that separate from this discussion.